R/colourInput.R

```diff
@@ -221,10 +221,16 @@ formatHEXsingle <- function(x) {
   if (x %in% grDevices::colors()) {
     x <- do.call(grDevices::rgb, as.list(grDevices::col2rgb(x) / 255))
   }
-  if (!grepl("^#?([[:xdigit:]]{3}|[[:xdigit:]]{6})$", x)) {
+  if (!grepl("^#?([[:xdigit:]]{3}|[[:xdigit:]]{6}|[[:xdigit:]]{8})$", x)) {
     stop(sprintf("%s is not a valid colour", x), call. = FALSE)
   }
 
+  # check whether it is 6-digit hex code with alpha.
+  if (substr(x, 1, 1) == "#" && nchar(x) == 9) {
```

Comment:
At least this is what I get with this code:

```
> plot(rnorm(100), col = "#FFF")
Error in plot.xy(xy, type, ...) : invalid RGB specification
```
Same using `"#FFFF"`. It works with `#FFFFFF` or `#FFFFFFFF`. I would have to check the R reference guide to make sure but it seems that it is not a valid specification. If it is not really legal, then you could simplify the code by removing the expansion at the end of the function.